examples/dart/test/geospatial_data_test.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:realm_dart/realm.dart';
+import 'package:test/test.dart';
+import 'utils.dart';
+
+part 'geospatial_data_test.g.dart';
+
+// :snippet-start: define-geopoint-class
+// To store geospatial data, create an embedded object with this structure.
+// Name it whatever is most convenient for your application.
+@RealmModel(ObjectType.embeddedObject)
+class _MyGeoPoint {
+  // These two properties are required to persist geo data.
+  final String type = 'Point';
+  final List<double> coordinates = const [0, 0];
+
+  // You can optionally implement convenience methods to simplify
+  // creating and working with geospatial data.
+  double get lon => coordinates[0];
+  set lon(double value) => coordinates[0] = value;
+
+  double get lat => coordinates[1];
+  set lat(double value) => coordinates[1] = value;
+
+  GeoPoint toGeoPoint() => GeoPoint(lon: lon, lat: lat);
+}
+// :snippet-end:
+
+// :snippet-start: use-geopoint-class
+// Then use it as a property in your model.
```

Comment:
this comment reads oddly in the doc since it's separated from the above piece. maybe tweak slightly or remove? 
```suggestion
// Use the GeoJSON-compatible class as a property in your model.
```